tools/config.py

```diff
@@ -797,6 +798,13 @@ def validate_config(self):
         return True
 
 
+    @property
+    def name(self):
+        if "artifact_name" in self.app_config_data:
```

Comment:
Also, agreed: we should discuss how naming from configs should affect the library name. If we come up with something, I will submit another PR that implements that thing that we came up with.